lib/ansible/modules/cloud/vmware/vmware_guest_network.py

```diff
@@ -0,0 +1,489 @@
+            e1000=vim.vm.device.VirtualE1000(),
+            e1000e=vim.vm.device.VirtualE1000e(),
+            sriov=vim.vm.device.VirtualSriovEthernetCard(),
+        )
+        if device_type and device_type in nic_device_type:
+            return nic_device_type[device_type]
+        else:
+            self.module.fail_json(msg='Invalid network device_type %s' % device_type)
+
+    @staticmethod
+    def get_parent_datacenter(obj):
+        """ Walk the parent tree to find the object's datacenter """
+        datacenter = None
+        while True:
+            if not hasattr(obj, 'parent'):
+                break
+            obj = obj.parent
+            if isinstance(obj, vim.Datacenter):
+                datacenter = obj
+                break
+        return datacenter
+
+    def network_exists_by_name(self, content, name):
+        """ Check if network with specified name exists """
+        vimtype = [vim.Network]
+        result = find_obj(content, vimtype, name)
+        if result:
+            if to_text(self.get_parent_datacenter(result).name) != to_text(self.params['datacenter']):
+                objects = get_all_objs(content, vimtype)
+                for obj in objects:
+                    if to_text(self.get_parent_datacenter(obj).name) == to_text(self.params['datacenter']):
+                        if to_text(obj.name) == to_text(name):
+                            return True
+            else:
+                return True
+        return False
+
+    def get_network_device_by_mac(self, vm=None, mac=None):
+        """ Get network adapter with the specified mac address"""
+        nic_device = None
+        if vm is None or mac is None:
+            return nic_device
+        for device in vm.config.hardware.device:
+            if isinstance(device, vim.vm.device.VirtualEthernetCard):
+                if device.macAddress == mac:
+                    nic_device = device
+                    break
+
+        return nic_device
+
+    def get_network_devices_by_type(self, vm=None, device_type=None):
+        """ Get network adapter list with the name type """
+        nic_devices = []
+        if vm is None or device_type is None:
+            return nic_devices
+        for device in vm.config.hardware.device:
+            if isinstance(device, self.nic_device_type[device_type]):
+                nic_devices.append(device)
+
+        return nic_devices
+
+    def get_network_device_by_label(self, vm=None, device_label=None):
+        """ Get network adapter with the specified label """
+        nic_device = None
+        if vm is None or device_label is None:
+            return nic_device
+        for device in vm.config.hardware.device:
+            if isinstance(device, vim.vm.device.VirtualEthernetCard):
+                if device.deviceInfo.label == device_label:
+                    nic_device = device
+                    break
+
+        return nic_device
+
+    @staticmethod
+    def is_valid_mac_addr(mac_addr):
```

Comment:
Can this moved to `module_utils/vmware.py` so that it can be used by other modules which requires it ?

Reply:
Yeah, sure, I saw somewhere also need this. Thank you.